csharp/sample/Microsoft.ML.OnnxTraining.MNISTSample/BinaryFile.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) 2018-2020 SignalPop LLC, and contributors. All rights reserved.
+// License: Apache 2.0
+// License: https://github.com/MyCaffe/MyCaffe/blob/master/LICENSE
+// Modified from Original Source: https://github.com/MyCaffe/MyCaffe/blob/master/MyCaffe.data/MnistDataLoader.cs
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+/// <summary>
+/// The MyCaffe.data namespace contains dataset creators used to create common testing datasets such as MNIST, CIFAR-10 and VOC0712.
+/// </summary>
+namespace MyCaffe.data
+{
+    /// <summary>
+    /// The BinaryFile class is used to manage binary files used by the MNIST dataset creator.
+    /// </summary>
+    class BinaryFile : IDisposable
+    {
+        FileStream m_file;
+        BinaryReader m_reader;
+
+        /// <summary>
+        /// The constructor.
+        /// </summary>
+        /// <param name="strFile">Specifies the filename.</param>
+        public BinaryFile(string strFile)
+        {
+            m_file = File.Open(strFile, FileMode.Open, FileAccess.Read, FileShare.Read);
+            m_reader = new BinaryReader(m_file);
+        }
+
+        /// <summary>
+        /// Release all resources.
+        /// </summary>
+        public void Dispose()
+        {
+            m_reader.Close();
+        }
+
+        /// <summary>
+        /// Returns the binary reader used.
+        /// </summary>
```

Comment:
>/// </summary> [](start = 8, length = 14)

<value> is missing

Reply:
Could you clarify?

Reply:
Functions require <return> documented. Properties require <value> documented that is returned. We would like examples to be good

---
In reply to: [544534423](https://github.com/microsoft/onnxruntime/pull/6141#discussion_r544534423) [](ancestors = 544534423)

Reply:
I agree.  What I am unclear on is what parameter/method is missing the comment.  From below, my understanding is that you are talking about this method:

        /// <summary>
        /// Returns the binary reader used.
        /// </summary>
        public BinaryReader Reader
        {
            get { return m_reader; }
        }

Is that correct?

Reply:
This is a property that needs to document return value which is done by <value></value> XML tag.

---
In reply to: [544583240](https://github.com/microsoft/onnxruntime/pull/6141#discussion_r544583240) [](ancestors = 544583240)

Reply:
I thought the <summary> tag does that already for a property.  The <returns> tag does not seem to apply - at least the VS editor does not seem to like it for a property.